weave/dataset/dataset.py

```diff
@@ -140,6 +140,8 @@ def add_rows(self, rows: Iterable[dict]) -> "Dataset":
         Returns:
             The updated dataset.
         """
+        import weave
```

Comment:
why do we need to import here?

Reply:
weave.publish is at the top level -- doing this for convenience while I refactor the other parts